src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -933,7 +930,22 @@ void System.Collections.IList.Remove(Object item)
             if (_size - index < count)
                 ThrowHelper.ThrowArgumentException(ExceptionResource.Argument_InvalidOffLen);
             Contract.EndContractBlock();
-            Array.Reverse(_items, index, count);
+
+            // The non-generic Array.Reverse is not used because it does not perform
+            // well for non-primitive value types.
+            // If/when a generic Array.Reverse<T> becomes available, the below code
+            // can be deleted and replaced with a call to Array.Reverse<T>.
+            int i = index;
+            int j = index + count - 1;
+            while (i < j)
+            {
+                T temp = _items[i];
```

Comment:
It may be beneficial to cache _items in a local variable - the JIT is likely to produce better code for it.